tensorcircuit/templates/ensemble.py

```diff
@@ -0,0 +1,150 @@
+        Expected to be run after finishing compile
+        """
+        if not self.permit_train:
+            # raise Exception("Needed to be compiled before training")
+            raise ValueError()
+        for i in range(self.count):
+            if "verbose" in kwargs:
+                if kwargs["verbose"] == 1:
+                    print("Model ", i + 1, "/", self.count, " is training...")
+            else:
+                print("Model ", i + 1, "/", self.count, " is training...")
+            self.__train_model(i, **kwargs)
+
+    def compile(self, **kwargs: kwargus) -> None:
+        self.permit_train = True
+        for i in range(self.count):
+            self.models[i].compile(**kwargs)
+
+    def __get_confidence(self, model_index: int, input: NDArray) -> NDArray:
+        """
+        Get the confidence value that is needed by voting.
+        Number of calling this function is reduced by self.need_confidence
+        """
+        self.need_confidence = False
+        prediction = self.models[model_index].predict(input, verbose=0)
+        prediction_returns = np.zeros(len(prediction))
+        for i in range(len(prediction)):
+            prediction_returns[i] = prediction[i][0]
+        return prediction_returns
+
+    """
+    Voting strategies begin
+    More voting strategies can be added beneath, a single function, and a if function in self.predict
+    """
+
+    def __voting_weight(self, array: NDArray) -> NDArray:
+        result = []
+        for i in array:
+            result.append(self.__voting_weight_single(i))
+        return np.array(result)
+
+    def __voting_average(self, array: NDArray) -> NDArray:
+        result = np.mean(array, axis=1)
+        return result
+
+    def __voting_weight_single(self, array: NDArray) -> float:
+        opp_array = np.ones(len(array)) - array
+        weight = np.absolute(opp_array - array)
+        weight_sum = np.sum(weight)
+        weight = weight / weight_sum
+        result = array * weight
+        return float(np.sum(result))
+
+    def predict(self, input_data: NDArray, voting_policy: str = "None") -> NDArray:
+        """
+        Input data is expected to be a 2D array that the first layer is different input data (into the trained models)
+        """
+        if self.need_confidence:
+            predictions = []
+            for i in range(self.count):
+                predictions.append(np.array(self.__get_confidence(i, input_data)))
+            self.predictions = np.transpose(np.array(predictions))
+        if voting_policy == "weight":
+            return self.__voting_weight(self.predictions)
+        elif voting_policy == "most":
+            return self.__voting_average(self.predictions)
+        elif voting_policy == "average":
+            return self.__voting_average(self.predictions)
+        elif voting_policy == "None" or voting_policy == "none":
```

Comment:
None is an object in python, not "None"

Reply:
The default of `voting_policy` is set to `none`, so I believe it is not possible for `voting_policy` to equals to None.

Reply:
No, what I mean is that the default policy should be ``None`` instead of ``"None"``